### ISSUE TITLE
An bug in the ElleKit code injection framework causes AppSync Unified to not work correctly on some jailbreak environments (※ Dopamine, palera1n-c in `-l` rootless mode, etc.) in certain configurations

### ISSUE THREAD

**Issue by jetblackrx89** [akemin-dayo/AppSync#174]
Multiple people over in the [Reddit post](https://www.reddit.com/r/jailbreak/comments/14otear/update_appsync_unified_1120_now_with_support_for/) announcing version 112 reported this issue, so I am formally filing a bug since it seems none of them did. We are getting a "Couldn't communicate with a helper application" error with the following combination:

- iPadOS 16.5.1
- Palera1n 2.0 beta 7
- Ellekit tweak injector installed
- Installing .IPA via Filza

It seems one person also claims to have his this on iPadOS 15.1 using Dopamine, so maybe the above aren't strict requirements. I also tried loading a .IPA directly without Filza via a site that hooks directly into AppSync Unified (if installed) and the app silently failed to install.

**Comment by EB2k05** [comment]
I’m having the exact same issue. iPad Pro 10.5”, iOS 16.5.1, palera1n, same error on Filza. It didn’t work with the command line tweak either.

**Comment by JEmlay** [comment]
I never even thought to try that.  Can you specify your dev account or will it endlessly resign?

**Comment by onlyzyz** [comment]
iPhone 8 iOS 16.5.1c palen1x rootless Jailbreak, install Appsync(Couldn't communicate with a helper application), then Trollhelper and TrollStore, now it works!

**Comment by EB2k05** [comment]
> I found you can install .ipa's by installing "TrollStore Helper" via Sileo/Zebra which in turn installs TrollStore. It should be available from the default repos and will self-sign the app. Even though the original exploit it worked on was patched, if you're already jailbroken, the signing still works fine.
> 
> Note that this has nothing to do with AppSync whatsoever. AppSync still has issues in terms of hooking into Filza and web-hosted .ipa links.

OMG it worked!! I thought Trollstore wasn’t available for iOS 16 but I guess it is. Thanks!

**Comment by marco-calautti** [comment]
I have the exact same issue: IPad 7th Gen (Cellular) on iPadOS 16.5.1 and palera1n rootless. Filza says couldn't communicate with helper application, and appinst on the command line simply says it failed installing. This happens with every IPA i tried.

**Comment by marco-calautti** [comment]
Using TrollStore as suggested allows to install unsigned IPAs.

**Comment by JEmlay** [comment]
I doubt this is a problem with AppSync.  Filza can't even properly install DEB files on rootless.  With AppSync I can install IPAs via sideloadly with my dev account.

For Filza you need to wait for them to fix their mess.

**Comment by marco-calautti** [comment]
As I said, the problem occurs also with the appinst command, which is provided by the AppSync author himself, and thus I would expect to work properly, but it doesn't.

**Comment by JEmlay** [comment]
And what does that have to do with Filza?  I’m not required to address your entire comment.  I addressed Filza.  Your follow up wasn’t needed.

**Comment by marco-calautti** [comment]
My follow-up is indeed needed, because it proves that the install issue is related to AppSync and not Filza. Your argument from before claimed that probably the problem is not in AppSync, since Filza has many issues. The fact that also appinst, the basic cli tool provided by the author to install IPAs has the same issue means the issue is not exclusive to Filza.

There is no need to be so passive-aggressive btw. Relax.

**Comment by JEmlay** [comment]
It proves no such thing and appsync works outside of filza.  Filza can’t install anything in rootless, regardless of appsync.

Your problem is with appinst.

Relax yourself kid.

**Comment by marco-calautti** [comment]
If even the most basic tool such as appinst, that uses AppSync to install IPAs, and which comes from the AppSync author himself has issues, then sorry, but I still tend to believe the problem is in AppSync itself, or at least in the way apps communicate with it.

Anyway, there is no point in keeping this discussion going. I will wait for a proper technical answer.

**Comment by JEmlay** [comment]
Good, quit yapping and wait.  Meanwhile my comment which you attempted to negate and failed is that appsync works outside of filza and filza has nothing to do with anything and should never be used as a metric for something else working or not.

**Comment by marco-calautti** [comment]
On my iPad AppSync does NOT work even OUTSIDE Filza, this is what I am trying to say since the beginning, but you are purposely ignoring what I am saying. Also the OP points out that AppSync does not work even outside Filza. So the discussion about Filza was over already after my first comment. It is just you that keep talking about that. I am not talking about Filza. AppSync simply does not work (even outside Filza) as pointed out multiple times.

**Comment by JEmlay** [comment]
Get over it.

**Comment by akemin-dayo** [comment]
Apologies for the late reply to this issue thread in general — I've been quite busy with a deadline for something that I had to attend to after I released the latest version of ASU.

Thank you everyone, for all the reports.

---

## What happened here?

During pre-release testing, I had attributed the `appinst` IPC failure behaviour that was presenting _only_ on the Dopamine jailbreak to something being broken in some bizarre way specific to the Dopamine environment that I was unable to find the root cause of at the time (and had the aforementioned deadline coming up).

The thing about `appinst` is that it is… an _extremely_ simple utility with not much room for error — it was utterly _incomprehensible_ to me what was even breaking as all it really does is call native Apple APIs to trigger app installation by way of `LSApplicationWorkspace` on iOS >= 8, or `MobileInstallation` on anything lower. In other words, it's just what iOS natively already does to begin with.

My best guess at the time was that the IPC failure was occurring either due to some kind of sandboxing issue, some kind of entitlements issue, or some other bizarre quirk regarding Dopamine that I simply did not understand. I _did_ try isolating that second possibility — just to make sure I hadn't gone _completely_ insane, I stripped all the entitlements off of the `appinst` binary once during testing just to observe the behaviour, and the failure mode was completely different… so, that meant entitlements _were_ working, as far as I could tell…

After days of repeatedly smashing my head into my keyboard, in the interest of allowing users with compatible configurations to use ASU as quickly as possible, I decided to release the updated versions of both ASU and `appinst` as-is, with an explicit warning regarding the known issues with Dopamine left in various places in the documentation, as seen below.

> ※ appinst does NOT currently work with the Dopamine jailbreak due to an IPC issue on that specific jailbreak. [[Twitter](https://twitter.com/akemin_dayo/status/1672982839405723651)] [[Fediverse (Mastodon, Misskey, etc.)](https://main.elk.zone/mstdn.jp/@akemin_dayo@mstdn.jp/110605445960091417)] [[Bluesky](https://bsky.app/profile/akemin-dayo.akemi.ai/post/3jyypxpxxpk25)]

After release, @opa334 (the developer of Dopamine) decided to take a closer look at what was happening, and discovered that at least on their configuration, `appinst` itself was fine, and AppSync Unified was actually the one that was somehow just… completely broken!?

> I looked into it for a bit and it seems appinst works fine on Dopamine, but AppSync is broken to the point where not even App Store app installations work. Disabling the installd hook fixes this so there must be something wrong with it. Looked at source code but cannot see what's wrong :/. It could be an ellekit bug but then I don't see why it would work on palera1n rootless.
> 
> EDIT: There seems to be a null pointer dereference in installd but for some reason instead of crashing it just spins forever:
> 
> ```gdb
> (lldb) thread select 2
> * thread #2, queue = 'com.apple.root.utility-qos'
>     frame #0: 0x0000000199eb7490 libobjc.A.dylib`objc_retain + 16
> libobjc.A.dylib`objc_retain:
> ->  0x199eb7490 <+16>: ldr    x10, [x9, #0x20]
>     0x199eb7494 <+20>: tbz    w10, #0x2, 0x199eb74d8    ; <+88>
>     0x199eb7498 <+24>: tbz    w8, #0x0, 0x199eb74f8     ; <+120>
>     0x199eb749c <+28>: mov    x9, #0x100000000000000
> (lldb) reg read
> General Purpose Registers:
>         x0 = 0x0000000104a163a0
>         x1 = 0x00000001ce2c966e
>         x2 = 0x0000000104a163a0
>         x3 = 0x00000001dc74fb98  @"Apple Inc."
>         x4 = 0x0000000104716750
>         x5 = 0x000000010472a580
>         x6 = 0x99a0ff7499d2056b
>         x7 = 0x00000001047307c0
>         x8 = 0x2000000000000000
>         x9 = 0x0000000000000000
> ```

In other words, the root cause of failure here at least on opa334's configuration appears to be with AppSync Unified itself and _not_ `appinst`.

I am inclined to agree with their deduction of the root cause simply because as I said, `appinst` is _so_ incredibly simple that I really cannot imagine what could have gone wrong there, other than the fabric of reality beginning to fray at the edges or something like that.

The weird thing is, during pre-release testing I _did_ try to isolate whether `appinst` itself or AppSync Unified was the root cause of the IPC failure. For one, I never observed any `installd` crashes. (Though given opa334's "it just spins forever" observation… yeah, that's not really a crash.)

Another thing I did was that I downloaded a FairPlay-encrypted IPA using `ipatool`, and tried to install it on a Dopamine system with only `appinst` installed and no ASU package.

And that… failed. _That's_ what led me to attribute the failure to `appinst` specifically instead of AppSync Unified. (That being said, I _did_ still add a "※ Users using the Dopamine jailbreak may encounter issues." warning for AppSync Unified before I released it regardless, just to err on the side of caution since it became clear that the very fabric of reality was breaking down before my eyes…)

Another thing of note is that I never actually encountered this peculiar "not even App Store app installations work" behaviour that opa334 mentioned. Had this been the case for me, it would have _definitely_ led me to discover that AppSync Unified was the root cause of the issue.

I even had a dev version of AppSync Unified that spewed some debug logging once it was injected into `installd`, and that worked just fine. Why and how opa334's Dopamine environment differs from mine in this regard, I have not yet figured out.

---

## The rootless-mode (`-l`) palera1n-c reports

All that being said… the various reports I've seen (both in this thread and on reddit) of AppSync Unified (and `appinst`) not working correctly on _some_ **rootless-mode** (`-l` argument) palera1n-c configurations are… unexpected, and actually did _not_ occur with my very limited pre-release testing of this particular combination.

(※ I do not personally own an iOS 16 device that can be used with palera1n-c, and thus had to resort to remote testing. At some point soon I will properly fully set up and post my Throne account on all my relevant SNS platforms so I can hopefully crowdfund a D221 or other palera1n-c-able device for development purposes so I can properly test locally.)

The fact that _some_ rootless-mode palera1n-c configurations are also exhibiting the same failure mode makes me feel that, as opa334 had suggested, this may be an issue somewhere in ElleKit, as both Dopamine and rootless-mode palera1n-c use ElleKit as their code injection framework. It's the closest thing that links both jailbreak environments, and it would also make sense as not many other things could cause this kind of failure. (※ Rooted/rootful/fakefs-mode palera1n-c does not use ElleKit.)

Gods know I've broken both Substrate and Substitute with my code (including ASU) in the past, might as well add ElleKit to the list of code injection frameworks I've somehow managed to utterly break. ;P

I will spend some time in the coming week looking into this, and probably will poke Évelyne to see if she has any idea as to what's going on.

Do note that it's _also_ possible the root cause might not even be ElleKit (it's simply just the most likely scenario right now with the information we currently have), but rather something else entirely.

We'll see.

And again, thank you all for reporting!

---

## Yeah, I… uh… I'm not reading all that.

**tl;dr:** Sorry for the late reply, and thank you for all the reports. My best guess at the cause of the issue as of this writing is that this is likely an issue with the ElleKit code injection framework. I will look into it and ask Évelyne if she has any idea as to what's going on.

**Comment by marco-calautti** [comment]
Thanks a lot for your detailed reply! You are doing a great job with ASU. I am looking forward for updates on this!

**Comment by marco-calautti** [comment]
If it can help, when installing palera1n in rootful mode, ASU works both using appinst, as well as other installers using ASU.

**Comment by akemin-dayo** [comment]
> If it can help, when installing palera1n in rootful mode, ASU works both using appinst, as well as other installers using ASU.

@marco-calautti Yes, this is expected behaviour — the rooted/rootful/fakefs (`-f`) version of palera1n-c does not use ElleKit, and has no other known issues that came up during pre-release testing. Thanks for reporting, though!

**Comment by hackcatml** [comment]
> You don't need a dev account. You just point Trollstore to an .ipa hosted in the Files app, and it installs it.
> 
> Again though, I would really like to see AppSync properly fixed since a lot of things are designed to work with it.

Hey, I had the same issue (iPhoneX iOS 16.5.1, palera1n rootless jb, ellekit 1.0)
I made a gist for it
https://gist.github.com/hackcatml/f7b7e0458df04e9bdd13583490cf0be1
You can check it out if you still want to install an .ipa using Filza
You need to build AppSync yourself, though

**Comment by opa334** [comment]
I looked into this now, this is the full backtrace I got from the crash:
```
    frame #0: 0x0000000198e27490 libobjc.A.dylib`objc_retain + 16
    frame #1: 0x000000010408e174 installd`___lldb_unnamed_symbol1143 + 72     -[MICodeSigningInfo initWithSignerIdentity:signerOrganization:codeInfoIdentifier:teamIdentifier:signatureVersion:entitlements:signerType:profileType:signingInfoSource:]
    frame #2: 0x00000001040ae684 installd`___lldb_unnamed_symbol1474 + 3996   -[MICodeSigningVerifier performValidationWithError:]
    frame #3: 0x0000000104085250 installd`___lldb_unnamed_symbol1033 + 1896   -[MIExecutableBundle codeSigningInfoByValidatingResources:performingOnlineAuthorization:ignoringCachedSigningInfo:checkingTrustCacheIfApplicable:skippingProfileIDValidation:error:]
    frame #4: 0x00000001040841f0 installd`___lldb_unnamed_symbol1029 + 68     -[MIExecutableBundle needsDataContainer]
    frame #5: 0x0000000104053dac installd`___lldb_unnamed_symbol525 + 80      +[MILaunchServicesDatabaseGatherer entryForBundle:inContainer:withError:]
    frame #6: 0x0000000104054fcc installd`___lldb_unnamed_symbol533 + 324     -[MILaunchServicesDatabaseGatherer scanExecutableBundle:inContainer:withError:]
    frame #7: 0x00000001040771d4 installd`___lldb_unnamed_symbol846 + 488     -[MIFilesystemScanner _scanBundleContainerType:withError:ignoredErrorOccurredOnOneOrMoreItems:]
    frame #8: 0x0000000104077484 installd`___lldb_unnamed_symbol847 + 92      -[MIFilesystemScanner _scanBundleContainersWithError:ignoredErrorOccurredOnOneOrMoreItems:]
    frame #9: 0x00000001040775b4 installd`___lldb_unnamed_symbol850 + 136     -[MIFilesystemScanner performScanWithError:]
    frame #10: 0x00000001040558d0 installd`___lldb_unnamed_symbol536 + 324    -[MILaunchServicesDatabaseGatherer performGatherWithError:]
    frame #11: 0x000000010405b974 installd`___lldb_unnamed_symbol623 + 996    sub_100013590
```

The issue is the following: `-[MICodeSigningVerifier performValidationWithError:]` calls `SecCertificateCopySubjectSummary`, which is hooked by AppSync, that hook seems to return garbage which is later passed to `-[MICodeSigningInfo initWithSignerIdentity:signerOrganization:codeInfoIdentifier:teamIdentifier:signatureVersion:entitlements:signerType:profileType:signingInfoSource:]` as the first argument, that method attempts to call `objc_retain` on it, which crashes the process (or makes it get stuck on Ellekit 1.0 apparently, but on 1.1 beta it crashes).

**Comment by opa334** [comment]
```
(lldb) disassemble -n SecCertificateCopySubjectSummary
Security`SecCertificateCopySubjectSummary:
    0x1895c673c <+0>:   cbz    x0, 0x1895c6858           ; <+284>
    0x1895c6740 <+4>:   pacibsp
    0x1895c6744 <+8>:   sub    sp, sp, #0x60
    0x1895c6748 <+12>:  stp    x22, x21, [sp, #0x30]
    0x1895c674c <+16>:  stp    x20, x19, [sp, #0x40]
    0x1895c6750 <+20>:  stp    x29, x30, [sp, #0x50]
```

The issue is that that `SecCertificateCopySubjectSummary` has a `cbz` before the function prologue, this causes ellekit to incorrectly hook it in a way where the call to orig will skip over the entire function and just return the value that just so happens to be in x0. As an experiment I have manually added +4 to the `SecCertificateCopySubjectSummary` function pointer. That solved it and both appinst and AppSync now work fine (Although only adhoc signed IPAs will run as Dopamine doesn't add anything non-adhoc-signed to trustcache).

**Comment by mhdj14** [comment]
@akemin-dayo Something confuses me about your explanation.

I’m currently using an iPhone 8 Plus, palera1n rootful jb AND using Ellekit, as Substrate/Substitute crashes my phone out of the jb continuously when using checkl0ck and a couple other tweaks, while Ellekit works with EVERYTHING.

**Comment by akemin-dayo** [comment]
@mhdj14 The issue is as opa334 said above — there's a bug in ElleKit (at least, on some devices/configurations…?) that results in a process crash when a specific C function, [`SecCertificateCopySubjectSummary()`](https://developer.apple.com/documentation/security/1396041-seccertificatecopysubjectsummary?language=objc) is hooked at all (※ even if you simply return the original implementation's behaviour).

The only thing you or I can do at this point is to remain patient and wait for Évelyne to determine the root cause of the issue whenever she has the free time to do so, and fix it in an update to ElleKit.

**Comment by evelyneee** [comment]
[debs.zip](https://github.com/akemin-dayo/AppSync/files/12836124/debs.zip)

Hey! I fixed these issues, and the installd issue isn't present anymore on my device. Could some of you test these debs (rootful & rootless) and confirm, they're zipped because GitHub won't upload debs.

**Comment by akemin-dayo** [comment]
I'm [busy travelling with some close friends](https://twitter.com/akemin_dayo/status/1709410790883885418) at the moment, but I will test this as soon as possible!

Thank you for your hard work!!

**Comment by JEmlay** [comment]
Hopefully this leads to Apple File Conduit getting fix.  SSH is such a nightmare and because every SSH app out there doesn't have proper entitlements, no access to app folders.

**Comment by akemin-dayo** [comment]
I have a full rewrite of `afc2d-arm64` that will be released… hopefully soon. It should be more reliable and fix all known issues with the current implementation.

**Comment by JEmlay** [comment]
Thank you so very much!

**Comment by bakirfatah99** [comment]
Work fine for me dopamine jailbreak rootless. But after install apps. App will crash. How get fix this crash?

**Comment by bakirfatah99** [comment]
> [debs.zip](https://github.com/akemin-dayo/AppSync/files/12836124/debs.zip)
> 
> 
> 
> Hey! I fixed these issues, and the installd issue isn't present anymore on my device. Could some of you test these debs (rootful & rootless) and confirm, they're zipped because GitHub won't upload debs. 

Work fine for me dopamine jailbreak rootless. But after install apps. App will crash. How get fix this crash?

**Comment by w2599** [comment]
> [debs.zip](https://github.com/akemin-dayo/AppSync/files/12836124/debs.zip)
> 
> Hey! I fixed these issues, and the installd issue isn't present anymore on my device. Could some of you test these debs (rootful & rootless) and confirm, they're zipped because GitHub won't upload debs.

It causes nfcd to keep crashing

**Comment by evelyneee** [comment]
Do you have Aemulo? Also can you send the log

**Comment by w2599** [comment]
> Do you have Aemulo? Also can you send the log

[nfcd-2023-10-07-193856.zip](https://github.com/akemin-dayo/AppSync/files/12837635/nfcd-2023-10-07-193856.zip)

**Comment by evelyneee** [comment]
[Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
Can you test these @w2599? Should fix the nfcd issue. 

I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?

And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side

**Comment by bakirfatah99** [comment]
> [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
> 
> Can you test these @w2599? Should fix the nfcd issue. 
> 
> 
> 
> I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> 
> 
> 
> And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side



I test it crash again app

**Comment by bakirfatah99** [comment]
I think appsync have problem

**Comment by opa334** [comment]
Apps will still require to be adhoc signed to work. This is not an AppSync problem.

**Comment by w2599** [comment]
> Can you test these @w2599? Should fix the nfcd issue. 
> 

Yes, nfcd has been fixed.

**Comment by Jaimeduenas47856** [comment]
> [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip) Can you test these @w2599? Should fix the nfcd issue.
> 
> I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> 
> And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side

This new Ellekit Beta causes Bioprotect to crash the SpringBoard

**Comment by Jaimeduenas47856** [comment]
This happens with both versions Ellekit 1.0-18.8c90474 and Ellekit 1.0-19.21b2b09

**Comment by admin-elvistrujanovic** [comment]
Here's installation error logs (I hope this can help you guys and girls to find an fix that doesn't require completely wiping all data everytime when AppSync is installed and after that removed)


For some reason, the damages like this stays even after removing the jailbreak completely:

[mobile_installation.log.0](https://github.com/akemin-dayo/AppSync/files/12872118/mobile_installation.log.0.txt)

**Comment by opa334** [comment]
> > [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
> > Can you test these @w2599? Should fix the nfcd issue.
> > I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> > And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side
> 
> I've downloaded Archive.zip from your repo and tested it
> 
> I can for sure says now that it showing me an error saying unable to generate serialized.... and every type of sideloading stopped to work - even from App Store
> 
> After I removed AppSync, the issue is still there
> 
> I was forced to reset my phone to fabric settings to recover it
> 
> So, AppSync is still so buggy and not working with Ellekit (requires more works)

this bug has nothing to do with appsync. It's unknown what causes it but the only known fix is to make a backup, wipe the device and restore the backup.

**Comment by admin-elvistrujanovic** [comment]
> > > [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
> 
> > > Can you test these @w2599? Should fix the nfcd issue.
> 
> > > I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> 
> > > And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side
> 
> > 
> 
> > I've downloaded Archive.zip from your repo and tested it
> 
> > 
> 
> > I can for sure says now that it showing me an error saying unable to generate serialized.... and every type of sideloading stopped to work - even from App Store
> 
> > 
> 
> > After I removed AppSync, the issue is still there
> 
> > 
> 
> > I was forced to reset my phone to fabric settings to recover it
> 
> > 
> 
> > So, AppSync is still so buggy and not working with Ellekit (requires more works)
> 
> 
> 
> this bug has nothing to do with appsync. It's unknown what causes it but the only known fix is to make a backup, wipe the device and restore the backup.

Well maybe not AppSync but installing this AppSync with latest test version of Ellekit could make some permanent bugs on iOS 15.4.1 arm64 like losing ability to sideload/install apps even in unjailbroken state.


Everything worked perfect before I installed Ellekit and after that AppSync


So, looking on the possible ways how this bug can happen and survive reboots would be very interesting


Maybe I don't know, I didn't try it yet but maybe re-flashing the current firmware with shsh2 blobs with option --update could keep my data and maybe try to regenerate everything about installd to start working again

**Comment by admin-elvistrujanovic** [comment]
So... I don't recommend arm64 users with iOS 15.4.1 to try this before an stable release comes out and also AppSync too

**Comment by opa334** [comment]
> > > > [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
> > 
> > 
> > > > Can you test these @w2599? Should fix the nfcd issue.
> > 
> > 
> > > > I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> > 
> > 
> > > > And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side
> > 
> > 
> > > 
> > 
> > 
> > > I've downloaded Archive.zip from your repo and tested it
> > 
> > 
> > > 
> > 
> > 
> > > I can for sure says now that it showing me an error saying unable to generate serialized.... and every type of sideloading stopped to work - even from App Store
> > 
> > 
> > > 
> > 
> > 
> > > After I removed AppSync, the issue is still there
> > 
> > 
> > > 
> > 
> > 
> > > I was forced to reset my phone to fabric settings to recover it
> > 
> > 
> > > 
> > 
> > 
> > > So, AppSync is still so buggy and not working with Ellekit (requires more works)
> > 
> > 
> > this bug has nothing to do with appsync. It's unknown what causes it but the only known fix is to make a backup, wipe the device and restore the backup.
> 
> Well maybe not AppSync but installing this AppSync with latest test version of Ellekit could make some permanent bugs on iOS 15.4.1 arm64 like losing ability to sideload/install apps even in unjailbroken state.
> 
> Everything worked perfect before I installed Ellekit and after that AppSync
> 
> So, looking on the possible ways how this bug can happen and survive reboots would be very interesting
> 
> Maybe I don't know, I didn't try it yet but maybe re-flashing the current firmware with shsh2 blobs with option --update could keep my data and maybe try to regenerate everything about installd to start working again

This has happen to people at random and I don't think AppSync could cause it. Some file permissions are corrupted causing installd to fail installing stuff, I never could figure out which permissions or why it happens. I already gave you the instructions to fix it - wipe device in settings and restore backup afterwards... futurerestore will not work.

**Comment by admin-elvistrujanovic** [comment]
> > > > > [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip)
> > > 
> > > 
> > > > > Can you test these @w2599? Should fix the nfcd issue.
> > > 
> > > 
> > > > > I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> > > 
> > > 
> > > > > And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > I've downloaded Archive.zip from your repo and tested it
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > I can for sure says now that it showing me an error saying unable to generate serialized.... and every type of sideloading stopped to work - even from App Store
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > After I removed AppSync, the issue is still there
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > I was forced to reset my phone to fabric settings to recover it
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > So, AppSync is still so buggy and not working with Ellekit (requires more works)
> > > 
> > > 
> > > this bug has nothing to do with appsync. It's unknown what causes it but the only known fix is to make a backup, wipe the device and restore the backup.
> > 
> > Well maybe not AppSync but installing this AppSync with latest test version of Ellekit could make some permanent bugs on iOS 15.4.1 arm64 like losing ability to sideload/install apps even in unjailbroken state.
> > 
> > Everything worked perfect before I installed Ellekit and after that AppSync
> > 
> > So, looking on the possible ways how this bug can happen and survive reboots would be very interesting
> > 
> > Maybe I don't know, I didn't try it yet but maybe re-flashing the current firmware with shsh2 blobs with option --update could keep my data and maybe try to regenerate everything about installd to start working again
> 
> This has happen to people at random and I don't think AppSync could cause it. Some file permissions are corrupted causing installd to fail installing stuff, I never could figure out which permissions or why it happens. I already gave you the instructions to fix it - wipe device in settings and restore backup afterwards... futurerestore will not work.

Well it's basically crazy how this can happen always to me when I install Ellekit + AppSync on rootless jb. This never happened to me in rootful and every day this worked for me - since today when I tried to install it


I've tried to give installd all permissions and for group and others also the same where I found installd-related directories


Anyway, I'm looking for a fix without data loss - if some update in Ellekit and AppSync cannot fix this mess - well, there is no other choice for me except to wipe the device (factory reset)

**Comment by evelyneee** [comment]
> > [Archive.zip](https://github.com/akemin-dayo/AppSync/files/12837975/Archive.zip) Can you test these @w2599? Should fix the nfcd issue.
> > I don't know why the apps crash - but it seemed it wasn't from signing issues on my side, so maybe AppSync has an issue?
> > And @akemin-dayo have a lovely time in Canada, it's really nice here. No rush on my side
> 
> This new Ellekit Beta causes Bioprotect to crash the SpringBoard

Can you send the crash log

**Comment by MuhammadBoubes** [comment]
> I'm [busy travelling with some close friends](https://twitter.com/akemin_dayo/status/1709410790883885418) at the moment, but I will test this as soon as possible!
> 
> 
> 
> Thank you for your hard work!!

Hi, I tested the fixed ellekit version on palera1n rootless (iphone x, ios 16.5): all IPAs I tried get installed but only few of them work, others crash immediately when opened...

Could this be related to what opa334 mentioned about "adhoc-signed IPAs" or this has nothing to do with palera1n jailbreak?

Have a nice holiday...

**Comment by opa334** [comment]
> > I'm [busy travelling with some close friends](https://twitter.com/akemin_dayo/status/1709410790883885418) at the moment, but I will test this as soon as possible!
> > Thank you for your hard work!!
> 
> Hi, I tested the fixed ellekit version on palera1n rootless (iphone x, ios 16.5): all IPAs I tried get installed but only few of them work, others crash immediately when opened...
> 
> Could this be related to what opa334 mentioned about "adhoc-signed IPAs" or this has nothing to do with palera1n jailbreak?
> 
> Have a nice holiday...

On palera1n ad-hoc signatures shouldn't be required. That's only a requirement for Dopamine.

**Comment by evelyneee** [comment]
@Jaimeduenas47856 Please send the Bioprotect crash log, or at least your device/iOS version ....

**Comment by evelyneee** [comment]
If someone else could try Bioprotect and report back, would be sooo appreciated since I might not get a response

**Comment by MuhammadBoubes** [comment]
> If someone else could try Bioprotect and report back, would be sooo appreciated since I might not get a response

I tested it on my device(iphone x, ios 16.5, palera1n rootless with ellekit 1.0-18.8c90474): the tweak needs to be purchased from settings, 
but nothing crashed and device didn't enter safe mode and I didn't find any crash logs related to bioprotect...

**Comment by mezii** [comment]
Any one solved this issue ?
I tried hooking into a root app (creating a root app by chmod +s /Applications/XXX.app/XX)  and it did not work. 
But when I do not set chmod +s (for permission) , i can be able hooking into the app. In the mean while, I can hook normally to root app when using with mobilesubstrate !

**Comment by evelyneee** [comment]
@mezii That’s not relevant to ElleKit. Also, this issue should be closed, it was fixed months ago @akemin-dayo

**Comment by opa334** [comment]
Not only that but the message should be removed, I might make a PR soon

**Comment by akemin-dayo** [comment]
No need for the PR — the message was already removed in my working source tree. (Been unexpectedly busy the past few months, things have only started to calm down recently o)-<)

Closing this issue thread.
